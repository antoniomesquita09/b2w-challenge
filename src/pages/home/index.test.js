import React from 'react'
import { useSelector } from 'react-redux'
import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import { MemoryRouter as Router } from 'react-router-dom'

import Home from './index.js'

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}))

describe('<Home />', () => {
  beforeEach(() => {
    useSelector.mockImplementation((cb) =>
      cb({
        pokemons: {
          allPokemons: [
            {
              id: 1,
              name: 'pikachu',
              price: 12,
            },
          ],
        },
        cart: {
          cart: [
            {
              id: 1,
              name: 'pikachu',
              price: 12,
            },
          ],
          total: 12,
        },
      })
    )
  })

  afterEach(() => {
    useSelector.mockClear()
  })

  it('Snapshot testing Home', () => {
    const { asFragment } = render(
      <Router>
        <Home />
      </Router>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
