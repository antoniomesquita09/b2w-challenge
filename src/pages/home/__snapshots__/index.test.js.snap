// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> Snapshot testing Home 1`] = `
<DocumentFragment>
  <div
    class="root"
    style="background-color: rgb(85, 89, 92); color: rgb(255, 255, 255);"
  >
    <h1>
      Pokémon Store
    </h1>
    <h2
      class="subTitle"
    >
      Tipo: 
      <span>
        normal
      </span>
    </h2>
  </div>
  <div
    class="root"
  >
    <input
      class="input"
      placeholder="Procure o seu pokemon pelo nome"
      value=""
    />
  </div>
  <div
    class="root"
  >
    <div>
      <ul>
        <li
          class="root"
          style="background-color: rgb(191, 193, 195);"
        >
          <h1>
            pikachu
          </h1>
          <h2>
            Preço: 
            <strong>
              R$12,00
            </strong>
          </h2>
          <svg
            class="icon"
            color="#c00"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(204, 0, 0);"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M22.73 22.73L2.77 2.77 2 2l-.73-.73L0 2.54l4.39 4.39 2.21 4.66-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h7.46l1.38 1.38c-.5.36-.83.95-.83 1.62 0 1.1.89 2 1.99 2 .67 0 1.26-.33 1.62-.84L21.46 24l1.27-1.27zM7.42 15c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h2.36l2 2H7.42zm8.13-2c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H6.54l9.01 9zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2z"
            />
          </svg>
        </li>
      </ul>
      <div
        class="root"
      >
        <button
          class="button"
          style="background-color: dodgerblue;"
        >
          <span>
            1
          </span>
        </button>
      </div>
    </div>
    <div
      class="root"
    >
      <h1>
        Carrinho
      </h1>
      <div
        class="item"
      >
        <h2
          data-testid="pokemon-name"
        >
          pikachu
        </h2>
        <h3>
          R$12,00
        </h3>
      </div>
      <h2>
        Total: R$12,00
      </h2>
      <button>
        Finalizar
      </button>
    </div>
  </div>
</DocumentFragment>
`;
